torch/onnx/_internal/fx/registration.py

```diff
@@ -192,83 +75,143 @@ def _initiate_registry_from_torchlib(
+                leave it to None.
         Returns:
-            The _SymbolicFunctionGroup object corresponding to the given name, or None if the name is not in the registry.
+            A list of SymbolicFunctions corresponding to the given name, or None if
+            the name is not in the registry.
         """
-        return self._registry.get(name)
+        internal_name_class = OpName.from_name_parts(
+            namespace=namespace, op_name=op_name, overload=overload
+        )
+        if (functions := self._registry.get(internal_name_class)) is not None:
+            return functions
+        return None
 
     @_beartype.beartype
-    def is_registered_op(self, name: str, version: int) -> bool:
-        """Returns whether the given op is registered for the given opset version.
+    def is_registered_op(
+        self, namespace: str, op_name: str, overload: Optional[str] = None
+    ) -> bool:
+        """Returns whether the given op is registered: torch.ops.<namespace>.<op_name>.<overload>.
 
         Args:
-            name: The qualified name of the function to check.
-            version: The opset version to check.
+            namespace: The namespace of the operator to check.
+            op_name: The name of the operator to check.
+            overload: The overload of the operator to check. If it's default overload,
+                leave it to None.
 
         Returns:
-            True if the given op is registered for the given opset version, otherwise False.
+            True if the given op is registered, otherwise False.
         """
-        functions = self.get_function_group(name)
-        if functions is None:
-            return False
-        return functions.get(version) is not None
+        functions = self.get_functions(
+            namespace=namespace, op_name=op_name, overload=overload
+        )
+        return functions is not None
 
     @_beartype.beartype
-    def all_functions(self) -> Set[str]:
+    def _all_registered_ops(self) -> Set[str]:
         """Returns the set of all registered function names."""
-        return set(self._registry)
+        return {
+            op_name_class.qualified_name() for op_name_class in self._registry.keys()
+        }
+
+
+@dataclasses.dataclass(frozen=True, eq=True)
+class OpName:
+    """A class representing an operator name in internal ONNX converter."""
+
+    namespace: str
+    op_name: str
+    overload: str
+
+    @classmethod
+    @_beartype.beartype
+    def from_name_parts(
+        cls, namespace: str, op_name: str, overload: Optional[str] = None
+    ) -> OpName:
+        # NOTE: in PyTorch, the overload could be unprovided to indicate the
+        # default overload
+        if overload is None or overload == "":
+            overload = "default"
+        return cls(namespace, op_name, overload)
+
+    @classmethod
+    @_beartype.beartype
+    def from_full_name(cls, full_name: str) -> OpName:
```

Comment:
nit: full_name and qualified_name now seems to mean the same thing, consider keeping only one.

Reply:
Keeping qualified_name